motionplan/plan.go

```diff
@@ -0,0 +1,154 @@
+package motionplan
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	pb "go.viam.com/api/service/motion/v1"
+	"go.viam.com/rdk/motionplan/ik"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+type Plan struct {
+	Trajectory
+	Path
+
+	// nodes corresponding to inputs can be cached with the Plan for easy conversion back into a form usable by RRT
+	// depending on how the Trajectory is constructed these may be nil and should be computed before usage
+	nodes []node
+}
+
+func newPlan(solution []node, sf *solverFrame, relative bool) (*Plan, error) {
+	traj := sf.nodesToTrajectory(solution)
+	path, err := newPath(solution, sf)
+	if err != nil {
+		return nil, err
+	}
+	return &Plan{
+		Trajectory: traj,
+		Path:       path,
+		nodes:      solution,
+	}, nil
+}
+
+// TODO: can probably think of a better name for this function
+func (plan *Plan) Remaining(waypointIndex int) *Plan {
```

Comment:
Perhaps `Tail` as a better name?

Reply:
I don't really like this one either.  Maybe `Subplan` or `Subsection`?

Reply:
I am putting in my vote for `RemainingPlan`
the name is very very clear about what the function does
`SubPlan` makes me think I am interested in the sub-steps between waypoints, as does `Subsection`.

Reply:
I think `RemainingPlan` is pretty good going to go with this.